File: C++/144_Binary Tree Preorder Traversal.cpp
class Solution {
public:
    vector<int> preorderTraversal(TreeNode *root) {
        v.clear();
        preorder(root);
        return v;
    }
private:
    vector<int> v;
    void preorder(TreeNode *root) {
        if (root == NULL)
            return;
        v.push_back(root->val);
        preorder(root->left);
        preorder(root->right);
    }
};

File: Python/144_Binary Tree Preorder Traversal.py
class Solution:
    # @param root, a tree node
    # @return a list of integers
    def preorderTraversal(self, root):
        res = []
        self._preOrder(root, res)
        return res

    def _preOrder(self, root, res):
        if not root:
            return
        res.append(root.val)
        self._preOrder(root.left, res)
        self._preOrder(root.right, res)